src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java

```diff
@@ -63,28 +80,46 @@ public String getConfirmation(@PathVariable String companyNumber,
             return ERROR_VIEW;
         }
 
-
         // If the payment is anything but paid return user to beginning of journey
+        PayableLateFilingPenalty payableLateFilingPenalty;
+        CompanyProfileApi companyProfileApi;
+        try {
+            companyProfileApi = companyService.getCompanyProfile(companyNumber);
+            payableLateFilingPenalty = payablePenaltyService
+                    .getPayableLateFilingPenalty(companyNumber, penaltyId);
+        } catch (ServiceException ex) {
+            LOGGER.errorRequest(request, ex.getMessage(), ex);
+            return ERROR_VIEW;
+        }
+
         if (!paymentStatus.equals("paid")) {
             LOGGER.info("Payment status is " + paymentStatus + " and not of status 'paid', returning to beginning of journey");
-            PayableLateFilingPenalty payableLateFilingPenalty;
-            try {
-                payableLateFilingPenalty = payablePenaltyService
-                        .getPayableLateFilingPenalty(companyNumber, penaltyId);
-            } catch (ServiceException ex) {
-                LOGGER.errorRequest(request, ex.getMessage(), ex);
-                return ERROR_VIEW;
-            }
-
             Map<String, String> links = payableLateFilingPenalty.getLinks();
             return UrlBasedViewResolver.REDIRECT_URL_PREFIX + links.get("resume_journey_uri");
 
         }
 
-        model.addAttribute("companyNumber", companyNumber);
-        model.addAttribute("penaltyNumber", penaltyId);
+        model.addAttribute(COMPANY_NUMBER_ATTR, companyNumber);
+        model.addAttribute(PENALTY_NUMBER_ATTR, penaltyId);
+        model.addAttribute(COMPANY_NAME_ATTR, companyProfileApi.getCompanyName());
+        model.addAttribute(REASON_ATTR, LATE_FILING_PENALTY_REASON);
+        model.addAttribute(PAYMENT_DATE_ATTR, setUpPaymentDateDisplay(payableLateFilingPenalty));
+        model.addAttribute(PENALTY_AMOUNT_ATTR, setUpPaymentAmountDisplay(payableLateFilingPenalty));
 
         return getTemplateName();
+    }
+
+    private String setUpPaymentDateDisplay(PayableLateFilingPenalty payableLateFilingPenalty) {
+        return (payableLateFilingPenalty.getPayment() == null
+                || payableLateFilingPenalty.getPayment().getPaidAt() == null) ?
+                null : LocalDateTime.parse(payableLateFilingPenalty.getPayment().getPaidAt(),
+                        DateTimeFormatter.ofPattern("uuuu-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.UK))
+                .format(DateTimeFormatter.ofPattern("d MMMM uuuu", Locale.UK));
+    }
 
+    private String setUpPaymentAmountDisplay(PayableLateFilingPenalty payableLateFilingPenalty) {
+        return (payableLateFilingPenalty.getPayment() == null
+                || payableLateFilingPenalty.getPayment().getPaidAt() == null) ?
+                null : "£" + payableLateFilingPenalty.getPayment().getAmount() + " (no VAT is charged)";
```

Comment:
I'm not a fan of returning nulls. Can we return an empty string here?
`if (payableLateFilingPenalty.getPayment() == null) {
            return StringUtils.isEmpty(payableLateFilingPenalty.getPayment().getPaidAt()) ? "" :
                    "£" + payableLateFilingPenalty.getPayment().getAmount() + " (no VAT is charged)";
        }
        return "";`

Reply:
Update as suggested

Reply:
Probably not in scope for this ticket, but if anything we'd need to handle having these nulls on the payment fields and potentially log some errors/warnings. Like I say, that will be coming in future ticket but something to keep in mind. Also, not opposed to passing back null here as it will need a refactor but can leave it as Aaron suggested.